src/Rules/Variables/UnsetRule.php

```diff
@@ -69,6 +69,45 @@ private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
 			}
 
 			return $this->canBeUnset($node->var, $scope);
+		} elseif (
+			$node instanceof Node\Expr\PropertyFetch
+			&& $node->name instanceof Node\Identifier
+		) {
+			$type = $scope->getType($node->var);
+			foreach ($type->getObjectClassReflections() as $classReflection) {
+				if ($classReflection->isReadOnly() || $classReflection->isImmutable()) {
```

Comment:
I'd completely leave out this section about readonly/immutable classes. It will be caught by the section about native properties which is okay by me.